### PR TITLE
trivial: fu-util: show a better error for non-responsive daemon

### PR DESCRIPTION
**Pull request by superm1**
See #2212 for more context.

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/hughsie/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-util.c [hunk 1]
```diff
@@ -2545,12 +2545,20 @@ fu_util_check_daemon_version (FuUtilPrivate *priv, GError **error)
 	const gchar *daemon = fwupd_client_get_daemon_version (priv->client);
 
 	if (g_strcmp0 (daemon, SOURCE_VERSION) != 0) {
-		g_set_error (error,
-			     FWUPD_ERROR,
-			     FWUPD_ERROR_NOT_SUPPORTED,
-			     /* TRANSLATORS: error message */
-			     _("Unsupported daemon version %s, client version is %s"),
-			     daemon, SOURCE_VERSION);
+		if (daemon == NULL) {
```
Comment (hughsie) [inline src/fu-util.c:2548]: i think this should go before the `if (g_strcmp0 (daemon, SOURCE_VERSION`

File: src/fu-util.c [hunk 2]
```diff
@@ -2545,12 +2545,20 @@ fu_util_check_daemon_version (FuUtilPrivate *priv, GError **error)
...
+			g_set_error_literal (error,
+					     FWUPD_ERROR,
+					     FWUPD_ERROR_NOT_SUPPORTED,
+					     /* TRANSLATORS: error message */
+					     _("Unable to connect to daemon"));
```
Comment (hughsie) [inline src/fu-util.c:2553]: maybe `daemon` is too technical and not specific enough? maybe `fwupd service` or something like that?

File: src/fu-util.c (hunk 1 above)
Comment (superm1) [reply src/fu-util.c]: fixed

File: src/fu-util.c (hunk 2 above)
Comment (superm1) [reply src/fu-util.c]: I made it `service` on the logic that snap uses `snap.fwupd.fwupd.service` and regular uses `fwupd.service` and I don't want to have to show either in an error string.

**Pull request merged**